xdsl/dialects/scf.py

```diff
@@ -689,6 +697,107 @@ def parse(cls, parser: Parser) -> Self:
         return op
 
 
+@irdl_op_definition
+class IndexSwitchOp(IRDLOperation):
+    name = "scf.index_switch"
+
+    arg = operand_def(IndexType)
+    cases = attr_def(DenseArrayBase)
+
+    output = var_result_def(AnyAttr())
+
+    default_region = region_def("single_block")
+    case_regions = var_region_def("single_block")
+
+    def __init__(
+        self,
+        arg: Operation | SSAValue,
+        cases: DenseArrayBase,
+        default_region: Region,
+        case_regions: Sequence[Region],
+        result_types: Sequence[Attribute],
+        attr_dict: dict[str, Attribute] | None = None,
+    ):
+        attributes: dict[str, Attribute] = attr_dict or {}
+        attributes["cases"] = cases
+
+        super().__init__(
+            operands=(arg,),
+            attributes=attributes,
+            regions=(default_region, case_regions),
+            result_types=(result_types,),
+        )
+
+    def verify_(self) -> None:
+        if self.cases.elt_type != i64:
+            raise VerifyException("case values should have type i64")
+
+        if len(self.cases.data) != len(self.case_regions):
+            raise VerifyException(
+                f"has {len(self.case_regions)} case regions but {len(self.cases.data)} case values"
+            )
+
+        cases = self.cases.data.data
+        if len(set(cases)) != len(cases):
+            raise VerifyException("has duplicate case value")
+
+        def verify_region(region: Region, name: str):
+            yield_op = region.block.last_op
+            if not yield_op or not isinstance(yield_op, Yield):
+                raise VerifyException(f"expected region {name} to end with `scf.yield`")
+
+            if yield_op.operand_types != self.result_types:
+                raise VerifyException(
+                    f"region {name} returns values of types {yield_op.operand_types}"
+                    f"but expected {self.result_types}"
+                )
+
+        verify_region(self.default_region, "default")
+        for name, region in zip(cases, self.case_regions):
+            verify_region(region, str(name))
```

Comment:
is this necessary?

Reply:
The verification is a direct translation of the verification for the mlir operation. It's as necessary as any verification I guess

Reply:
I expect I could use `SingleBlockImplicitTerminator` and leave it at that?

Reply:
My impression is that the regions of the operation are already verified by the time the custom verifier is called, is that not the case? Can you check that they're not verified twice?

Reply:
`verify_region` is a custom function defined above

Reply:
Ah, I see, I thought we had `Operation.verify` call into helper methods on the operation, and that this was also doing this manually. In that case all good, although maybe it might be better as a private method?

Reply:
I should really be a bit more attentive in PRs, I'm not sure how I missed it. It's private indeed, but that just means that you define it every time that you verify this operation. Ideally, it should be a static method with an underscore, so that it's defined only once, when the file is imported.

Reply:
No problem, I'll make a quick pr